extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts

```diff
@@ -275,26 +278,24 @@ export class ToolsAndEulaPage extends ResourceTypePage {
 	}
 
 	private getCurrentProvider(): DeploymentProvider {
-		const options: { option: string, value: string }[] = [];
 
-		this._optionDropDownMap.forEach((selectBox, option) => {
-			let selectedValue: azdata.CategoryValue = selectBox.value as azdata.CategoryValue;
-			options.push({ option: option, value: selectedValue.name });
-		});
+		const options = this.getSelectedOption();
 
 		this.resourceProvider = this._resourceType.getProvider(options)!;
 		return this._resourceType.getProvider(options)!;
 	}
 
 	private getCurrentOkText(): string {
-		const options: { option: string, value: string }[] = [];
+		return this._resourceType.getOkButtonText(this.getSelectedOption())!;
+	}
 
+	private getSelectedOption(): { option: string, value: string }[] {
+		const options: { option: string, value: string }[] = [];
```

Comment:
`return this._optionDropDownMap.map((selectBox, option) => { return { option: option, value: selectedValue.name}; });`

Reply:
@Charles-Gagnon, it is a map and not an array.